packages/forklift-console-plugin/src/modules/Plans/data.ts

```diff
@@ -110,8 +111,8 @@ export const mergeData = (
               !vm.error &&
               !vm.conditions?.find((condition) => condition.type === 'Canceled'),
           ).length || 0,
-        migrationCompleted: migration?.completed,
-        migrationStarted: migration?.started,
+        migrationCompleted: convertToUTC(migration?.completed),
```

Comment:
we don't do that in other places, we don't need to do it here,
you make sure we use UTC here:
https://github.com/kubev2v/forklift-console-plugin/pull/747/files#diff-5c809c708698ad27265c0c3a2e2970700e439c4d826a6e4b651a67c4d8499a31R103

so no, need to do it twice

Reply:
1. the matcher is a re-usable component which additionally gets input from the user (the filter). That's why we need to repeat the conversions.
2.  Plans screen follows the "flat" design - the data is converted once and then used across UI. Since we decided to use UTC this is the right place to put the transformation. Note that we also perform validation here - invalid dates are discarded.

Reply:
> the matcher is a re-usable component which additionally gets input from the user (the filter). That's why we need to repeat the conversions.

This is the reason it's in the matcher, and I agree we should do a conversion in the matcher.
Here on the other hand it's wrong, it removed information we may want, e.g. the timezone.
We agreed to show users UTC when creating the new date filter, not to change the way we store data internally.  
This file is not part of the date filter, so if we want to change we can do it in a different PR, and IMHO we have no need to change the internal data, even in anther PR.

> Plans screen follows the "flat" design 

right, that will change in 2.6, but even using "flat" design we were fine with the data as is, since this code is not directly connected to the filter, I prefer to not touch it in this PR

Reply:
> This is the reason it's in the matcher, and I agree we should do a conversion in the matcher.
Here on the other hand it's wrong, it removed information we may want, e.g. the timezone.
>

The idea of the flattening approach was to perform data transformations once and let the "dumb" components (i.e. table cells) consume the results directly. If we want to support only UTC then this is the best place to put the logic.
Otherwise each component needs to remember to do the conversion.


>>Plans screen follows the "flat" design
>
> right, that will change in 2.6, but even using "flat" design we were fine with the data as is, since this code is not directly connected to the filter, I prefer to not touch it in this PR
>

The main reason of adding the column was to demo the filter (we have no other field to test it).

Reply:
> The main reason of adding the column was to demo the filter (we have no other field to test it).

You are removing `migrationStarted` and changing `migrationCompleted` 
I don't see an added column, am i missing something ?

My question is, why do you change  `migrationCompleted`  , AFAIU the demo of the date filter will work just the same with or without this change ?

p.s.
IFAIU the date and time data is always encapsulated in a component that will show it in the correct way,
once inside the filter, that does not show the data directly, and in the second place it's used as data to `Timestamp`  component that expect the unmodified data, see:
https://github.com/openshift/console/blob/5002947c029ad1dfdc1b0dc88275e8dea5670cf7/frontend/public/components/utils/details-page.tsx#L155

Reply:
> You are removing migrationStarted and changing migrationCompleted
>

Not removing but converting both `migrationStarted` and `migrationStarted` to UTC.

> I don't see an added column, am i missing something ?
>

Please check changes in the `PlansPage` -> https://github.com/kubev2v/forklift-console-plugin/pull/747/files#diff-e7a795084a21e23e501a782f72e5e5f90ba86bc27e9facbb1310fbce3d443b86

> My question is, why do you change migrationCompleted , AFAIU the demo of the date filter will work just the same with or without this change ?
>

As you requested the UI should be UTC based so converting both values is justified.

> IFAIU the date and time data is always encapsulated in a component that will show it in the correct way,
once inside the filter, that does not show the data directly, and in the second place it's used as data to Timestamp component that expect the unmodified data
>

`Timestamp` component will display any date. The objective was to display the date in UTC. This could be accomplished by converting in the component layer (in the table cell) or in the data transform layer.
As already pointed above the advantage of "flat" approach is that the data is pre-calculated once. This is used for other props as well i.e. check the `vmDone` prop:
```
vmDone:
          migration?.vms?.filter(
            (vm) =>
              !!vm.completed &&
              !vm.error &&
              !vm.conditions?.find((condition) => condition.type === 'Canceled'),
          ).length || 0,
```

Reply:
Our primary focus for this PR is creating the date filter component. To maintain clarity and focus, could we consider reverting the unrelated changes? This would help streamline our review and ensure alignment with the main objective. I appreciate your attention to this matter and your collaboration.

Reply:
> Is there any other part that you would like to revert?

I appreciate your input. For this particular instance, there's no need to make the changes you suggested in lines 113 and 114. Thank you for your understanding and cooperation.